src/modules/identity/RadixRemoteIdentity.ts

```diff
@@ -0,0 +1,226 @@
+            _socket = new WebSocket(this.remoteUrl)
+        }
+        return _socket
+    }
+
+    /**
+     * Signs an atom with the wallet using the remote identity
+     * 
+     * @param atom - The atom to be signed
+     * @returns A promise with the signed atom
+     */
+    public signAtom(atom: RadixAtom): Promise<RadixAtom> {
+        return new Promise<RadixAtom>((resolve, reject) => {
+            const socket = this.getSocketConnection(this.socket)
+
+            socket.onopen = () => {
+                socket.send(JSON.stringify({
+                    jsonrpc: '2.0',
+                    method: 'sign_atom',
+                    params: { 
+                        token: this.token,
+                        atom: atom.toJson(),
+                    },
+                    id: 0,
+                }))
+                socket.onmessage = (evt) => {
+                    const response = JSON.parse(evt.data)
+                    if (response.id === 0) {
+                        if (response.result) {
+                            atom.signatures = RadixSerializer.fromJson(response.result)
+                            resolve(atom)
+                        } else {
+                            reject(response.error)
+                        }
+                    }
+                }
+                socket.onerror = (error) => reject(new Error(JSON.stringify(error)))
+            }
+        })
+    }
+
+    /**
+     * Decrypt the payload of an atom
+     * 
+     * @param payload - The payload of the atom to be decrypted
+     * @returns A promise with the decrypted payload
+     */
+    public decryptECIESPayload(payload: Buffer): Promise<Buffer> {
+        return new Promise<Buffer>((resolve, reject) => {
+            const socket = this.getSocketConnection(this.socket)
+
+            socket.onopen = () => {
+                socket.send(JSON.stringify({
+                    jsonrpc: '2.0',
+                    method: 'decrypt_ecies_payload',
+                    params: {
+                        token: this.token,
+                        payload,
+                    },
+                    id: 1,
+                }))
+                socket.onmessage = (evt) => {
+                    const response = JSON.parse(evt.data)
+                    if (response.id === 1) {
+                        if (response.result && response.result.data) {
+                            resolve(response.result.data)
+                        } else {
+                            reject(response.error)
+                        }
+                    }
+                }
+                socket.onerror = (error) => reject(new Error(JSON.stringify(error)))
+            }
+        })
+    }
+
+    /**
+     * Returns the public key of this identity synchronously
+     * 
+     * @returns The public key of the identity
+     */
+    public getPublicKey(): Buffer {
+        return Buffer.from(this.keyPair.keyPair.getPublic().encode('be', true))
```

Comment:
The `getPublic()` method already returns a buffer

```
getPublic(): Buffer {
    return Buffer.from(this.keyPair.getPublic().encode('be', true))
}
```

```suggestion
        return this.keyPair.keyPair.getPublic()
```